resource/resource.go

```diff
@@ -51,6 +52,11 @@ type DesiredComposed struct {
 	Ready Ready
 }
 
+// ExtraResource is a resource requested by a Function.
+type ExtraResource struct {
```

Comment:
```suggestion
type Extra struct {
```

`resource.ExtraResource` stutters (kinda). I think just `Extra` is more in line with what we already have (e.g. `resource.Composite` not `resource.CompositeResource`).